Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/measure.h

```diff
@@ -532,11 +532,106 @@ volume(const TriangleMesh& tmesh, const CGAL_PMP_NP_CLASS& np)
+  * computes the centroid of the domain bounded by
+  * a closed triangulated surface mesh.
+  *
+  * @tparam TriangleMesh a model of `HalfedgeGraph`
+  * @tparam NamedParameters a sequence of \ref pmp_namedparameters "Named Parameters"
+  *
+  * @param tmesh the closed triangulated surface mesh bounding the volume
+  * @param np optional sequence of \ref pmp_namedparameters "Named Parameters" among the ones listed below
+  *
+  * @pre `tmesh` is closed
+  *
+  * \cgalNamedParamsBegin
+  *  \cgalParamBegin{vertex_point_map} the property map with the points associated to the vertices of `pmesh`.
+  *   If this parameter is omitted, an internal property map for
+  *   `CGAL::vertex_point_t` must be available in `TriangleMesh`\cgalParamEnd
+  *  \cgalParamBegin{geom_traits} an instance of a geometric traits class, model of `Kernel`\cgalParamEnd
+  * \cgalNamedParamsEnd
+  *
+  * @return the centroid of the domain bounded by `tmesh`.
+  */
+  template<typename TriangleMesh
+         , typename CGAL_PMP_NP_TEMPLATE_PARAMETERS>
+#ifdef DOXYGEN_RUNNING
+  Point_3
+#else
+  typename GetGeomTraits<TriangleMesh, CGAL_PMP_NP_CLASS>::type::Point_3
+#endif
+centroid(const TriangleMesh& tmesh, const CGAL_PMP_NP_CLASS& np)
+{
+  // See: http://www2.imperial.ac.uk/~rn/centroid.pdf
+    
+  CGAL_assertion(is_triangle_mesh(tmesh));
+  CGAL_assertion(is_closed(tmesh));
+
+  using boost::choose_param;
+  using boost::get_param;
+
+  typedef typename GetVertexPointMap<TriangleMesh, CGAL_PMP_NP_CLASS>::const_type Vpm;
+  Vpm vpm = choose_param(get_param(np, internal_np::vertex_point),
+                         get_const_property_map(CGAL::vertex_point, tmesh));
+  typedef typename GetGeomTraits<TriangleMesh, CGAL_PMP_NP_CLASS>::type Kernel;
+  typedef typename Kernel::Point_3 Point_3;
+  typedef typename Kernel::Vector_3 Vector_3;
+  typedef typename Kernel::Tetrahedron_3 Tetrahedron_3;
+  typedef typename Kernel::Construct_translated_point_3 Construct_translated_point_3;
+  typedef typename Kernel::Construct_vector_3 Construct_vector_3;
+  typedef typename Kernel::Compute_scalar_product_3 Scalar_product;
+  typedef typename Kernel::Construct_cross_product_vector_3 Cross_product;
+  typedef typename Kernel::Construct_scaled_vector_3 Scale;
+  typedef typename Kernel::Construct_sum_of_vectors_3 Sum;
+  typedef typename boost::graph_traits<TriangleMesh>::face_descriptor face_descriptor;
+
+  typename Kernel::FT volume = 0.;
+
+  Vector_3 centroid(NULL_VECTOR);
+
+  Construct_translated_point_3 point;
+  Construct_vector_3 vector;
+  Scalar_product scalar_product;
+  Cross_product cross_product;
+  Scale scale;
+  Sum sum;
+
+  for(face_descriptor fd : faces(tmesh)){
+    Vector_3 vp = vector(ORIGIN, get(vpm, target(halfedge(fd, tmesh), tmesh))),
+      vq = vector(ORIGIN, get(vpm, target(next(halfedge(fd, tmesh), tmesh), tmesh))),
+      vr = vector(ORIGIN, get(vpm, target(prev(halfedge(fd, tmesh), tmesh), tmesh)));
+    Vector_3 n = cross_product((vq-vp),(vr-vp));
```

Comment:
```suggestion
    Vector_3 n = cross_product(vector(vp,vq)),vector(vp,vr));
```

Reply:
No. The correct way uses ConstructSumOfVectors_3  and ConstructScaledVector_3  with FT(-1).

Reply:
uses `Construct_normal_vector_3` now.

Reply:
Missing `git push`?